ts/metrics/metric_cache_abstract.py

```diff
@@ -0,0 +1,462 @@
+"""
+Abstract metric cache class using ABC.
+Implemented in the case it is decided that another file format is better in the future.
+
+Currently, abstract class has the methods for getting a metric and adding a metric to the cache.
+"""
+import abc
+import logging
+
+import ts.metrics.metric_cache_errors as merrors
+from ts.metrics.dimension import Dimension
+from ts.metrics.metric import Metric
+from ts.metrics.metric_type_enums import MetricTypes
+
+
+class MetricCacheAbstract(metaclass=abc.ABCMeta):
+    def __init__(self, request_ids, model_name, file):
```

Comment:
I thought we didn't yet support a different metric cache per model because we load a single YAML file?

Also wondering should the metric cache also have a concept of a worker?